consensus/polybft/state_sync_manager.go

```diff
@@ -0,0 +1,492 @@
+			return nil, err
+		}
+
+		if err := s.initTransport(); err != nil {
+			return nil, err
+		}
+	*/
+
+	return s, nil
+}
+
+func (s *StateSyncManager) initTransport() error {
+	return s.topic.Subscribe(func(obj interface{}, _ peer.ID) {
+		msg, ok := obj.(*polybftProto.TransportMessage)
+		if !ok {
+			s.logger.Warn("failed to deliver message, invalid msg", "obj", obj)
+
+			return
+		}
+
+		var transportMsg *TransportMessage
+
+		if err := json.Unmarshal(msg.Data, &transportMsg); err != nil {
+			s.logger.Warn("failed to deliver message", "error", err)
+
+			return
+		}
+
+		if err := s.deliverMessage(transportMsg); err != nil {
+			s.logger.Warn("failed to deliver message", "error", err)
+		}
+	})
+}
+
+func (s *StateSyncManager) deliverMessage(msg *TransportMessage) error {
+	s.lock.Lock()
+	epoch := s.epoch
+	valSet := s.validatorSet
+	s.lock.Unlock()
+
+	if valSet == nil {
+		// Epoch metadata is undefined
+		return nil
+	}
+
+	if msg.EpochNumber < epoch {
+		// or received message for some of the older epochs.
+		return nil
+	}
+
+	if !valSet.validators.ContainsNodeID(msg.NodeID) {
+		return fmt.Errorf("validator is not among the active validator set")
+	}
+
+	msgVote := &MessageSignature{
+		From:      msg.NodeID,
+		Signature: msg.Signature,
+	}
+
+	numSignatures, err := s.state.insertMessageVote(msg.EpochNumber, msg.Hash, msgVote)
+	if err != nil {
+		return fmt.Errorf("error inserting message vote: %w", err)
+	}
+
+	s.logger.Info(
+		"deliver message",
+		"hash", hex.EncodeToString(msg.Hash),
+		"sender", msg.NodeID,
+		"signatures", numSignatures,
+	)
+
+	return nil
+}
+
+func (s *StateSyncManager) initTracker() error {
+	provider, err := jsonrpc.NewClient(s.jsonrpcAddr)
+	if err != nil {
+		return err
+	}
+
+	store, err := boltdbStore.New(filepath.Join(s.dataDir, "/deposit.db"))
+	if err != nil {
+		return err
+	}
+
+	s.logger.Info("Start tracking events", "bridge", s.bridgeAddr)
+
+	tt, err := tracker.NewTracker(provider.Eth(),
+		tracker.WithBatchSize(10),
+		tracker.WithStore(store),
+		tracker.WithFilter(&tracker.FilterConfig{
+			Async: false,
+			Address: []ethgo.Address{
+				ethgo.Address(s.bridgeAddr),
+			},
+		}),
+	)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		go func() {
+			if err := tt.Sync(context.Background()); err != nil {
+				s.logger.Error("Event tracker", "failed to sync", err)
+			}
+		}()
+
+		go func() {
+			for {
+				select {
+				case evnt := <-tt.EventCh:
+					if len(evnt.Removed) != 0 {
+						panic("this will not happen anymore after tracker v2")
+					}
+
+					for _, log := range evnt.Added {
+						fmt.Println("- log -")
+						fmt.Println(log)
+
+						if stateTransferEventABI.Match(log) {
+							if err := s.addLog(log); err != nil {
+								s.logger.Error("failed to decode state sync event", "hash", log.TransactionHash, "error", err)
+							}
+						}
+					}
+				case <-tt.DoneCh:
+					s.logger.Info("Historical sync done")
+				}
+			}
+		}()
```

Comment:
Bear in mind that this part of the PR (or possibly entire PR) is going to clash with @vcastellm, at least because of capturing `event_tracker` in a separate package: https://github.com/0xPolygon/polygon-edge/pull/1044/files#diff-4cd7f6ea48e9e952f59f2be59cf5142240f7d7337766f99b2ac04bb337775a82R17

Reply:
Used `eventTracker` from `tracker` package:

Fix: https://github.com/0xPolygon/polygon-edge/pull/1067/commits/503224004886fb793d1d63d972144d5323a9e546